lib/services/project-files-manager.ts

```diff
@@ -0,0 +1,56 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import path = require("path");
+import util = require("util");
+
+export class ProjectFilesManager implements IProjectFilesManager {
+	constructor(private $fs: IFileSystem,
+		private $platformsData: IPlatformsData) { }
+	
+	public processPlatformSpecificFiles(directoryPath: string, platform: string, excludedDirs?: string[]) {
+		return (() => {
+			var contents = this.$fs.readDirectory(directoryPath).wait();
+			var files: string[] = [];
+			
+			_.each(contents, fileName => {
+				let filePath = path.join(directoryPath, fileName);
+				let fsStat = this.$fs.getFsStats(filePath).wait();
+				if(fsStat.isDirectory() && (!excludedDirs || (excludedDirs && !_.contains(excludedDirs, fileName)))) {
```

Comment:
you do not need the !excludeDirs check - contains returns false if the first parameter is null/undefined:
```
> l.contains(undefined, "test")
false
> l.contains(null, "test")
false
```